clang/lib/CodeGen/TargetInfo.cpp

```diff
@@ -8372,10 +8372,15 @@ void TCETargetCodeGenInfo::setTargetAttributes(
 
         SmallVector<llvm::Metadata *, 5> Operands;
         Operands.push_back(llvm::ConstantAsMetadata::get(F));
-        ASTContext &Ctx = M.getContext();
-        unsigned XDim = Attr->getXDimVal(Ctx)->getZExtValue();
-        unsigned YDim = Attr->getYDimVal(Ctx)->getZExtValue();
-        unsigned ZDim = Attr->getZDimVal(Ctx)->getZExtValue();
+        const auto *XDimExpr = cast<ConstantExpr>(Attr->getXDim());
+        const auto *YDimExpr = cast<ConstantExpr>(Attr->getYDim());
+        const auto *ZDimExpr = cast<ConstantExpr>(Attr->getZDim());
+        Optional<llvm::APSInt> XDimVal = XDimExpr->getResultAsAPSInt();
+        Optional<llvm::APSInt> YDimVal = YDimExpr->getResultAsAPSInt();
+        Optional<llvm::APSInt> ZDimVal = ZDimExpr->getResultAsAPSInt();
+        unsigned XDim = XDimVal->getZExtValue();
+        unsigned YDim = YDimVal->getZExtValue();
+        unsigned ZDim = ZDimVal->getZExtValue();
```

Comment:
```suggestion
        unsigned XDim = Attr->getXDimVal()->getZExtValue();
        unsigned YDim = Attr->getYDimVal()->getZExtValue();
        unsigned ZDim = Attr->getZDimVal()->getZExtValue();
```
Any reason not to go with this shorter form?